src/core/cgroup.h

```diff
@@ -142,6 +143,7 @@ struct CGroupContext {
 
         char **ip_filters_ingress;
         char **ip_filters_egress;
+        LIST_HEAD(CGroupPinnedBPFProgram, pinned_bpf_progs);
```

Comment:
hmm, sofar CGroupContext was more of a declarative, pasive structure, that contained user settings but no runtime info. i.e. it's context is pretty much constant (except when the user reconfigures things). It carried no reference to runtime objects, i.e. fds and such.

all rutime objects we added to Unit and Service or whatever is the most appropriate. In aprticular all cgroup runtime objects are part of Unit. This one should be too hence.

Reply:
@poettering

Moved runtime info to Unit.
Have a concern regarding dbus getter, `property_get_bpf_foreign_program` : It doesn't have access to Unit, only to CGroupContext, and since with the most recent change change bpffs paths are not stored in CGroupContext, It's a stub getter merely opening and closing a container.